File: Server/SuperBoard/src/main/java/com/narara/superboard/cardlabel/interfaces/CardLabelAPI.java
package com.narara.superboard.cardlabel.interfaces;

import com.narara.superboard.cardlabel.interfaces.dto.CardLabelDto;
import com.narara.superboard.common.interfaces.response.DefaultResponse;
import com.narara.superboard.member.entity.Member;
import io.swagger.v3.oas.annotations.Operation;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequestMapping("/api/v1/card-label")
public interface CardLabelAPI {

    @PostMapping("/create")
    @Operation(summary = "카드 라벨 생성", description = "특정 카드에 라벨을 생성합니다.")
    ResponseEntity<DefaultResponse<CardLabelDto>> createCardLabel(
            @AuthenticationPrincipal Member member,
            @RequestBody Long cardId,
            @RequestBody Long labelId);

    @PatchMapping("/activate")
    @Operation(summary = "카드에 라벨 활성화 상태 변경", description = "카드 라벨의 활성화 상태를 변경합니다.")
    ResponseEntity<DefaultResponse<CardLabelDto>> changeCardLabelIsActivated(
            @AuthenticationPrincipal Member member,
            @RequestBody Long cardId,
            @RequestBody Long labelId);

    @GetMapping("/{cardId}")
    @Operation(summary = "카드 라벨 조회", description = "특정 카드에 속한 모든 라벨을 조회합니다.")
    ResponseEntity<DefaultResponse<List<CardLabelDto>>> getCardLabelCollection(@PathVariable Long cardId);

}

File: Android/core/network/src/main/java/com/ssafy/network/api/CardLabelAPI.kt
package com.ssafy.network.api

import com.ssafy.model.label.CreateCardLabelRequestDto
import com.ssafy.model.label.UpdateCardLabelActivateRequestDto
import com.ssafy.model.with.CardLabelDTO
import com.ssafy.network.source.ApiResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path

interface CardLabelAPI {
    @POST("api/v1/card-label/create")
    suspend fun createCardLabel(
        @Body createCardLabelRequestDto: CreateCardLabelRequestDto
    ): Response<ApiResponse<CardLabelDTO>>

    @GET("api/v1/card-label/{cardId}")
    suspend fun getCardLabels(
        @Path("cardId") cardId: Long,
    ): Response<ApiResponse<List<CardLabelDTO>>>

    @PATCH("api/v1/card-label/activate")
    suspend fun updateLabelActivate(
        @Body updateCardLabelActivateRequestDto: UpdateCardLabelActivateRequestDto
    ): Response<ApiResponse<CardLabelDTO>>
}